doc/reference/trd-appid.md

```diff
@@ -0,0 +1,787 @@
+contains a public 4096-bit RSA key (512 bytes), followed by a 512-byte
+signature that contains a 32-bit identifier. **Exact signature
+algorithm/approach to be determined.**
+
+The `SHA256` type has a data length of 32 bytes. It contains a 256-bit
+(32 byte) SHA256 hash of the application binary.
+
+The `SHA384` type has a data length of 48 bytes. It contains a 384-bit
+(48 byte) SHA384 hash of the application binary.
+
+The `SHA512` type has a data length of 64 bytes. It contains a 512-bit
+(64 byte) SHA512 hash of the application binary.
+
+`TbfFooterV2Credentials` follow the compiled app binary in a TBF
+object.  If a `TbfFooterV2Credentials` footer includes a cryptographic
+hash, signature, or other value to check the integrity of a process
+binary, this vlaue MUST be computed over the TBF Header and Userspace
+Binary, from the start of the TBF object until
+`binary_end_offset`. Computing an integrity value in a Credentials
+Footer MUST NOT include the contents of Footers. If new metadata
+associated with an application binary needs to be covered by
+integrity, it MUST be a Header. If new metadata associated with an
+application binary needs to not be covered by integrity, it MUST be a
+Footer.
+
+Which types of credentials a Credentials Checking Policy supports are
+kernel-specific. For example, an application that only accepts TBF
+Objects signed with a particular 4096-bit RSA key can support only
+`Rsa4096Key` credentials, while an open research system might support
+no credentials. Because the `length` field specifies the length of a
+given credentials, not understanding a particular credentials type
+does not prevent parsing others.
+
+
+6 Credentials Checking Policy: the `AppCredentialsChecker` trait
+===============================
+
+The `AppCredentialsChecker` trait defines the interface that implements
+the Credentials Checking Policy of the Process Checker: it accepts,
+passes on, or rejects Application Credentials. When a Tock board asks
+the kernel to load processes, it passes a reference to a
+`AppCredentialsChecker`, which the kernel uses to check credentials.
+An implementer of `AppCredentialsChecker` sets the security policy of
+Userspace Binary loading by deciding which types of credentials, and
+which credentials, are acceptable and which are rejected.
+
+
+```rust
+pub enum CheckResult {
+    Accept,
+    Pass,
+    Reject
+}
+
+pub trait Client<'a> {
+    fn check_done(&self,
+                  result: Result<CheckResult, ErrorCode>,
+                  credentials: TbfFooterV2Credentials,
+                  binary: &'a [u8]);
+}
+
+pub trait AppCredentialsChecker<'a> {
+    fn set_client(&self, client: &'a dyn Client<'a>);
+    fn require_credentials(&self) -> bool;
+    fn check_credentials(&self,
+                         credentials: TbfFooterV2Credentials,
+                         binary: &'a [u8])  ->
+        Result<(), (ErrorCode, TbfFooterV2Credentials, &'a [u8])>;
+}
+```
```

Comment:
Do we need this here? This just seems prone to bit rotting

Reply:
What do you mean? The point is to document the API, and then once the TRD is finalized the API doesn't change (unless the TRD is updated).